src/datasources/targeted-messaging/targeted-messaging.datasource.ts

```diff
@@ -101,6 +102,25 @@ export class TargetedMessagingDatasource
     return this.outreachDbMapper.map(dbOutreach);
   }
 
+  async getOutreach(outreachId: number): Promise<Outreach> {
+    const [dbOutreach] = await this.sql<
+      DbOutreach[]
+    >`SELECT target_all FROM outreaches WHERE id = ${outreachId}`.catch(
+      (err) => {
+        this.loggingService.warn(
+          `Error getting outreach: ${asError(err).message}`,
+        );
+        throw new UnprocessableEntityException('Error getting outreach');
+      },
+    );
+
+    if (!dbOutreach) {
+      throw new UnprocessableEntityException('Outreach not found');
```

Comment:
I think it would make sense to use a `NotFoundException` here.

Reply:
https://github.com/safe-global/safe-client-gateway/pull/2194/commits/65896bde7d8fef30ed73e817916f179ddf6cf334#diff-7b46005236ccadec6f3c236cb143960dcc7f8c53afb914e828c907dd112384dbL118